packages/docusaurus-plugin-pwa/src/index.js

```diff
@@ -88,12 +89,15 @@ function plugin(context, options) {
     injectHtmlTags() {
       const headTags = [];
       if (isProd && pwaHead) {
-        pwaHead.forEach(({tagName, ...attributes}) =>
-          headTags.push({
+        pwaHead.forEach(({tagName, ...attributes}) => {
+          if (attributes.href) {
+            attributes.href = normalizeUrl([baseUrl, attributes.href]);
```

Comment:
in this specific case it won't be a problem because it's only run during the production build

however I think plugin options should never be mutated by a plugin, and think it would be a good idea to use [deep-freeze](https://www.npmjs.com/package/deep-freeze) to ensure that in the future

Also for retro-compatibility I suggest not adding the baseurl if href already starts with the baseurl? It is unlikely the user will put pwa assets in `/baseUrl/baseUrl/manifest.json` anyway so it seems a safe thing to do

Reply:
> Also for retro-compatibility I suggest not adding the baseurl if href already starts with the baseurl? It is unlikely the user will put pwa assets in /baseUrl/baseUrl/manifest.json anyway so it seems a safe thing to do

Make sense, done it.